### PR TITLE
Add ddoc style comments to all diagnostic functions.

### PR DESCRIPTION
**Pull request by ibuclaw**
Mostly mechanical / repetitive comments.

### CODE REVIEW

File: src/dmd/errors.d [hunk 1]
```diff
@@ -72,6 +95,12 @@ extern (C++) void errorSupplemental(const ref Loc loc, const(char)* format, ...)
     va_end(ap);
 }
 
+/**
+ * Print a warning message, increasing the global warning count.
+ * Params:
+ *      loc = location of error
+ *      format = printf-style format specification
+ */
```
Comment (wilzbach) [inline src/dmd/errors.d]: Not that I would care much about (and it makes sense in DMD's ugly codebase), but as an FYI we recently defined documentation strings in the [DStyle](https://dlang.org/dstyle.html#phobos_documentation).

File: src/dmd/errors.d [hunk 2]
```diff
@@ -151,7 +202,16 @@ private void verrorPrint(const ref Loc loc, Color headerColor, const(char)* head
     fflush(stderr);     // ensure it gets written out in case of compiler aborts
 }
 
-// header is "Error: " by default (see errors.h)
+/**
+ * Same as $(D error), but takes a va_list parameter, and optionally additional message prefixes.
```
Comment (wilzbach) [inline src/dmd/errors.d]: FYI: Phobos is slowly converted to use backticks instead of `$(D ...)`

**Review by wilzbach — APPROVED** [review]
Doc improvements are always welcome!

**Comment by JinShil** [comment]
Per https://dlang.org/dstyle.html#phobos_documentation

> Documentation comments should not have leading stars or zeroes on each line.

But I don't think that should hold up this PR.

**Review by JinShil — APPROVED** [review]

File: src/dmd/errors.d (hunk 2 above)
Comment (ibuclaw) [reply src/dmd/errors.d]: Would you prefer that changed?

Comment (wilzbach) [reply src/dmd/errors.d]: It was just an FYI. I have given up hope for a pretty DMD codebase long ago.

**Comment by wilzbach** [comment]
> But I don't think that should hold up this PR.

Yes, it was just an FYI as it was added four days ago and I assume it's not very known: https://github.com/dlang/dlang.org/pull/2129

Anyhow the DDoc failures are a holdup:

```
dmd/errors.d(41): Warning: Ddoc: parameter count mismatch
dmd/errors.d(55): Warning: Ddoc: parameter count mismatch
dmd/errors.d(71): Warning: Ddoc: parameter count mismatch
dmd/errors.d(90): Warning: Ddoc: parameter count mismatch
dmd/errors.d(104): Warning: Ddoc: parameter count mismatch
dmd/errors.d(120): Warning: Ddoc: parameter count mismatch
dmd/errors.d(120): Warning: Ddoc: parameter count mismatch
dmd/errors.d(135): Warning: Ddoc: parameter count mismatch
dmd/errors.d(150): Warning: Ddoc: parameter count mismatch
```

**Comment by ibuclaw** [comment]
> Anyhow the DDoc failures are a holdup:

I didd have a quick look in phobos/druntime, and I couldn't see any existing documentation for ellipses.

**Comment by ibuclaw** [comment]
I'm not convinced that adding a `...` parameter works.

http://dtest.dlang.io/artifact/website-4add4e35db29d29eacfafcc97d5b06f9455e1982-10dea7bc7624f4925e8c3df676e15966/web/library-prerelease/dmd/errors/message.html

**Comment by JinShil** [comment]
I restarted Jenkins

File: src/dmd/errors.d [hunk 3]
```diff
@@ -80,6 +114,15 @@ extern (C++) void warning(const ref Loc loc, const(char)* format, ...)
     va_end(ap);
 }
 
+/**
+ * Print additional details about a warning message.
+ * Doesn't increase the error count or print an additional warning prefix.
+ * Params:
+ * Params:
```
Comment (jacob-carlborg) [inline src/dmd/errors.d]: Double params.

**Review by jacob-carlborg — CHANGES_REQUESTED** [review]

File: src/dmd/errors.d [hunk 4]
```diff
@@ -64,6 +83,14 @@ extern (C++) void error(const(char)* filename, uint linnum, uint charnum, const(
     va_end(ap);
 }
 
+/**
+ * Print additional details about an error mesage.
```
Comment (WalterBright) [inline src/dmd/errors.d]: message

**Comment by ibuclaw** [comment]
@WalterBright - I ran the source through aspell and picked up a couple of other spelling mistakes.

**Comment by ibuclaw** [comment]
Yep, although the compiler complains if `... `is missing, it doesn't do a good job formatting it.

http://dtest.dlang.io/artifact/website-6a94b9083d4ac94d03bfb3506e1cabbbfd3df19e-a80c86ba9803958733cdbcefd4eca848/web/library-prerelease/dmd/errors/deprecation.html

Or recognising it.

http://dtest.dlang.io/artifact/website-6a94b9083d4ac94d03bfb3506e1cabbbfd3df19e-a80c86ba9803958733cdbcefd4eca848/web/library-prerelease/dmd/errors.html

File: src/dmd/errors.d [hunk 5]
```diff
@@ -289,9 +392,9 @@ private void colorSyntaxHighlight(OutBuffer* buf)
 }
 
 
-/****************************
+/**
  * Embed these highlighting commands in the text stream.
- * HIGHLIGHT.Escape indicats a Color follows.
+ * HIGHLIGHT.Escape indicates a Color follows.
```
Comment (jacob-carlborg) [inline src/dmd/errors.d]: Backticks around `HIGHLIGHT.Escape`.

**Review by jacob-carlborg — APPROVED** [review]

File: src/dmd/errors.d [hunk 6]
```diff
@@ -21,7 +21,7 @@ import dmd.root.outbuffer;
 import dmd.root.rmem;
 import dmd.console;
 
-/**********************
```
Comment (WalterBright) [inline src/dmd/errors.d]: It's not really necessary to do this. The number of *'s is not consistent anywhere in the source code, and it hardly matters.

**Comment by ibuclaw** [comment]
> Yep, although the compiler complains if ...is missing, it doesn't do a good job formatting it.

Ah I've just realised that this is a ddox bug. The ddoc page is fine. :-)

**Pull request merged**